packages/core/src/labels.ts

```diff
@@ -26,8 +34,29 @@ export function withLabels<T>(operation: Operation<T>, labels: Labels): Operatio
  * Like {@link withLabels}, but replaces the existing labels entirely.
  */
 export function setLabels<T>(operation: Operation<T>, labels: Labels): Operation<T> {
-  if(operation) {
+  if (isObjectOperation<T>(operation)) {
+    let original = operation[Symbol.operation];
+    operation = {
+      ...labels,
+      [Symbol.operation]: original
+    } as Operation<T>;
+  } else if (operation) {
     operation.labels = labels;
   }
   return operation;
 }
+
+export function extractLabels<T>(operation: Operation<T>): Labels | undefined {
+  if (!isObjectOperation<T>(operation)) {
+    return operation?.labels;
```

Comment:
Do we need to also treat the name as a special case here?

Reply:
In in other words if this is not an `ObjectOperation` then both `name` and `labels` are fields that contain labels.

Reply:
@wKich ☝🏻